src/System.Memory/tests/BuffersExtensions/BuffersExtensionsTests.cs

```diff
@@ -29,6 +29,55 @@ public void WritingToMultiSegmentBuffer()
             Assert.Equal("Hello World!", bufferWriter.ToString());
         }
 
+        [Theory]
+        [InlineData(1, 0)]
+        [InlineData(10, 9)]
```

Comment:
Seems like you should have 0,0 as well.  That will be impacted by my previous suggestion/question about not having the first check you added.